projection.py

```diff
@@ -0,0 +1,28 @@
+import numpy as np
+
+def proj_i(i, n):
+    """Return the matrix for a projection operator, projecting onto the
+    vector i in the standard basis of an n-dim vector space.
+    """
+    a = np.zeros(n, dtype=np.complex256)
+    a[i] = 1
+    return np.outer(a, a)
+
+def psum(v, bit_to_measure):
+    """Return a projection operator that projects v onto the subspace spanned
+    by all basis vectors except v, i.e. V' = span({b: b in V and b!=v} where V
+    is the vector space containing v. The operator is formed as the sum of
+    projections on each basic vector b in V'.
+    """
+    n = len(v)
+    s = np.zeros((n, n), dtype=np.complex256)
+    for i in range(n):
+        if i == bit_to_measure:
+            continue
+        s += proj_i(i, n)
+    return s
```

Comment:
Don't we need to project down to all base vectors that contain the randomly selected state. For example `proj = |0000><0000| + |0100><0100|` if the random selection gave a 0 on qubit 0. 

Or if the random selection gave qubit values 1 and 0 we should have `proj = |0010><0010|`.